### PR TITLE
Implement GraphQL Connection Approach

### PR DESCRIPTION
**Pull request by ibrohimislam**
as on #1964. the connection approach implemented in `JpaStorageServices` class.

### CODE REVIEW

**Comment by ibrohimislam** [comment]
@jamesagnew, any advice to solve the error?

**Comment by jamesagnew** [comment]
hi @ibrohimislam - very very sorry to have missed this!!

yes I'm definitely happy to review, will do so now. :)

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/graphql/DaoRegistryGraphQLStorageServices.java [hunk 1]
```diff
@@ -228,10 +258,88 @@ public ReferenceResolution lookup(Object theAppInfo, IBaseResource theContext, I
...
+		int pageSize;
+		IBundleProvider response;
+
+		if (searchIdArgument.isPresent() && searchOffsetArgument.isPresent()) {
+			searchId = searchIdArgument.get();
+			searchOffset = Integer.parseInt(searchOffsetArgument.get());
+
+			response = Optional.ofNullable(myPagingProvider.retrieveResultList(requestDetails, searchId)).orElseThrow(()->{
+				String msg = myContext.getLocalizer().getMessageSanitized(DaoRegistryGraphQLStorageServices.class, "invalidGraphqlCursorArgument", searchId);
+				return new InvalidRequestException(msg);
```
Comment (jamesagnew) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/graphql/DaoRegistryGraphQLStorageServices.java:287]: This exception should have a code - Have a look at uses of `Msg.code` elsewhere to see how this works. You can use the javadoc in `Msg.java` to see what's the next available number, then make sure to update that javadoc

**Review by jamesagnew — APPROVED** [review]
Super cool!! One small comment, but happy to merge after that

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/graphql/DaoRegistryGraphQLStorageServices.java (hunk 1 above)
Comment (ibrohimislam) [reply hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/graphql/DaoRegistryGraphQLStorageServices.java:287]: on master, the error code 1277 is https://github.com/hapifhir/hapi-fhir/blob/5dfdc91682b09c07185c436bc1088acd57e437d5/hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/graphql/DaoRegistryGraphQLStorageServices.java#L234

Is it okay to reuse code 1277 for this? I don't know what last error code is.

**Review by jamesagnew — APPROVED** [review]

**Pull request merged**